packages/block-library/src/gallery/gallery.js

```diff
@@ -91,15 +100,11 @@ export const Gallery = ( props ) => {
 				} ) }
 			</ul>
 			{ mediaPlaceholder }
-			<RichText
-				tagName="figcaption"
-				className={ captionClassNames }
-				placeholder={ __( 'Write gallery caption…' ) }
-				value={ caption }
-				unstableOnFocus={ onFocusGalleryCaption }
-				onChange={ ( value ) => setAttributes( { caption: value } ) }
-				inlineToolbar
-			/>
+			{ ! isSelected && RichText.isEmpty( caption ) ? (
+				<VisuallyHidden>{ richTextElem }</VisuallyHidden>
```

Comment:
This presents a problem, because `<figcaption>` has to be a direct descendant of `<figure>`. I suspect the best way forward here will be to just pass the `components-visually-hidden` class to RichText directly. We have to support direct use of the class anyway because of the php files 🤷‍♀

Reply:
It's a bit mad, but this should work:
```jsx
<VisuallyHidden as={ RichText } tagName="figcaption" // other RichText props ... />
```

But looking at the VisuallyHidden implementation, it doesn't look like it accepts additional class names. I may knock together a quick PR to fix that.

Reply:
PR to better handle class names in VisuallyHidden:
https://github.com/WordPress/gutenberg/pull/20649

Reply:
Yes, let's fix it :) 

In general, you should be able to pass anything you want as `as` and all props should be passed down.

Reply:
@talldan I think that makes sense! I added a helper component inline since it's pretty specific to this file, and I prepared a more general helper in https://github.com/WordPress/gutenberg/pull/20928